benchmarks/headers.js

```diff
@@ -0,0 +1,147 @@
+const { Worker, isMainThread, parentPort, workerData } = require('worker_threads')
+
+function printResults (results, n) {
+  console.log('Results for Header append:')
+  const baselineTiming = Number.parseInt(results['undici-fetch'].endTime - results['undici-fetch'].startTime)
+  for (const [key, timing] of Object.entries(results)) {
+    const elapsedTT = Number.parseFloat(timing.endTime - timing.startTime)
+    console.log(`${key.padEnd(25)} | total time: ${elapsedTT}ns (${(elapsedTT * 0.000001).toFixed(3)}ms)`)
+  }
+  console.log('---')
+  for (const [key, timing] of Object.entries(results)) {
+    if (key === 'undici-fetch') continue
+    const elapsedTT = Number.parseFloat(timing.endTime - timing.startTime)
+    const percent = ((baselineTiming - elapsedTT) / elapsedTT) * 100
+    console.log(`undici-fetch <> ${key} percent change: ${percent.toFixed(3)}%`)
+  }
+}
+
+if (isMainThread) {
+  /** Sorted list of common header keys */
+  const commonHeaderKeys = [
+    'A-IM',
+    'Accept',
+    'Accept-Charset',
+    'Accept-Datetime',
+    'Accept-Encoding',
+    'Accept-Language',
+    'Accept-Patch',
+    'Accept-Ranges',
+    'Access-Control-Request-Headers',
+    'Access-Control-Request-Method',
+    'Age',
+    'Allow',
+    'Alt-Svc',
+    'Authorization',
+    'Cache-Control',
+    'Connection',
+    'Content-Disposition',
+    'Content-Encoding',
+    'Content-Language',
+    'Content-Length',
+    'Content-Location',
+    'Content-Range',
+    'Content-Type',
+    'Cookie',
+    'Date',
+    'Delta-Base',
+    'ETag',
+    'Expect',
+    'Expires',
+    'Forwarded',
+    'From',
+    'Host',
+    'IM',
+    'If-Match',
+    'If-Modified-Since',
+    'If-None-Match',
+    'If-Range',
+    'If-Unmodified-Since',
+    'Last-Modified',
+    'Link',
+    'Location',
+    'Max-Forwards',
+    'Origin',
+    'Pragma',
+    'Proxy-Authenticate',
+    'Proxy-Authorization',
+    'Public-Key-Pins',
+    'Range',
+    'Referer',
+    'Retry-After',
+    'Server',
+    'Set-Cookie',
```

Comment:
Set-Cookie can have multiple instances in a response. would be good to add a couple more here to exercise codepaths for dealing with non-unique headers

Reply:
good idea! I'll make sure to include this. 

Similarly, do you have any ideas for header values? Should I be testing something more than the single string included in the script already?

Reply:
almost all headers are strings with no post processing so it shouldn't really matter. On average your test string of `'standard-value'` is maybe a little short for the average header but that shouldn't matter too much for this kind of benchmark. I dont know enough about the internals to recommend using a number or a string representation of a number for things like `Content-Length` that are explicitly not arbitrary strings.

I will clone this branch and do a run with header specific strings for each key and see if it actually impacts the result. 
If there isn't a difference when using header specific values I wouldn't worry about it...

Reply:
yeah good point. I think maybe making `'standard-value'` longer will be the only tweak necessary; since you're right that most times headers are longer than that.